pkg/filesystem/local_directory_linux.go

```diff
@@ -34,3 +43,32 @@ func (d *localDirectory) Mknod(name path.Component, perm os.FileMode, deviceNumb
 func clonefileImpl(oldFD int, oldName string, newFD int, newName string) error {
 	return status.Error(codes.Unimplemented, "Clonefile is only supported on Darwin")
 }
+
+// Uses `unmount` on relative pathnames with `fchdir`.
+func fchdir_unmountat(dfd int, mountname string) error {
+	fchdirLock.Lock()
+	defer fchdirLock.Unlock()
+
+	store, err := os.Getwd()
+	if err != nil {
+		return util.StatusWrap(err, "Could not store the working directory %#v.")
+	}
+	defer os.Chdir(store)
+
+	err = syscall.Fchdir(dfd)
+	if err != nil {
+		err = &fs.PathError{Op: "fchdir", Path: fmt.Sprintf("%d", dfd), Err: err}
+		return util.StatusWrapf(err, "Could not change working directory to %#v", dfd)
+	}
+
+	err = unix.Unmount(mountname, 0)
+	if err != nil {
+		return util.StatusWrap(err, "Unmount")
+	}
+
+	return nil
+}
+
+func (d *localDirectory) Unmount(mountpoint path.Component) error {
+	return fchdir_unmountat(d.fd, mountpoint.String())
```

Comment:
We only use `fchdir_unmountat()` in one place. Be sure to just inline `Unmount()` into it.

Shouldn't you use `defer runtime.KeepAlive(d)` here as well, by the way? Otherwise you're at risk of having the directory closed before the `Unmount()` system call has had a chance to run.